search_service/proxy/elasticsearch.py

```diff
@@ -377,6 +376,130 @@ def fetch_table_search_results(self, *,
                                    query_name=query_name,
                                    model=Table)
 
+    @staticmethod
+    def convert_query_json_to_query_dsl(search_request: dict,
+                                        query_term: str) -> str:
+        """
+        Convert the generic query json to query DSL
+        e.g
+        ```
+        {
+            'type': 'AND'
+            'filters': {
+                'database': ['hive', 'bigquery'],
+                'schema': ['test-schema1', 'test-schema2'],
+                'table': ['*amundsen*'],
+                'column': ['*ds*']
+                'tag': ['test-tag']
+            }
+        }
+
+        This generic JSON will convert into DSL depending on the backend engines.
+
+        E.g in Elasticsearch, it will become
+        'database':('hive' OR 'bigquery') AND
+        'schema':('test-schema1' OR 'test-schema2') AND
+        'table':('*amundsen*') AND
+        'column':('*ds*') AND
+        'tag':('test-tag')
+        ```
+
+        :param search_request:
+        :return: The search engine query DSL
+        """
+        filter_list = search_request.get('filters')
+        add_query = ''
+        query_dsl = ''
+
+        if filter_list:
+            query_dsl = ' AND '.join([(category + ':' + '(' +
+                                     ' OR '.join(item_list) + ')')
+                                      for category, item_list in filter_list.items()])
```

Comment:
its nice that you got this all into one line, but I wonder if this logic would be more approachable if you made it a proper for loop and moved the replace logic below into the loop (replacing the items as you touch them rather than doing it after joining the string query together)

Reply:
Will let Tao comment

Reply:
yeah, I think the code is getting a bit hard to understand on how to construct the query dsl. If it is easy to change, I am all for making it easy to understand